mkosi/distributions/arch.py

```diff
@@ -0,0 +1,190 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+import os
+import platform
+from pathlib import Path
+from textwrap import dedent
+from typing import List, Set
+
+from ..backend import (
+    MkosiPrinter,
+    MkosiState,
+    OutputFormat,
+    add_packages,
+    complete_step,
+    disable_pam_securetty,
+    run,
+    sort_packages,
+)
+from ..mounts import mount_api_vfs
+from . import DistributionInstaller
+
+
+class ArchInstaller(DistributionInstaller):
+    @classmethod
+    def cache_path(cls) -> List[str]:
+        return ["var/cache/pacman/pkg"]
+
+    @classmethod
+    def install(cls, state: "MkosiState") -> None:
+        return install_arch(state)
```

Comment:
My thinking was, that `DistributionInstaller` would never have state beyond state that generally applies to that specific distribution - there's already examples in the Debian version - and state specific to our current invocation would be in `MkosiState`. 

I'd prefer to keep it like this for now and we can change it later, since changing it from a class to an instance method will always be easier.